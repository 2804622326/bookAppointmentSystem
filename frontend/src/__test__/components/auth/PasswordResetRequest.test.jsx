import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import PasswordResetRequest from '../../../components/auth/PasswordResetRequest.jsx';

jest.mock('../../../components/auth/AuthService.js', () => ({
  requestPasswordReset: jest.fn(() => Promise.resolve({ message: 'sent' })),
}));

jest.mock('../../../components/hooks/UseMessageAlerts', () => ({ __esModule: true, default: jest.fn() }));
import useAlerts from '../../../components/hooks/UseMessageAlerts';
const setShowSuccessAlert = jest.fn();
const setSuccessMessage = jest.fn();
const setErrorMessage = jest.fn();
const setShowErrorAlert = jest.fn();
useAlerts.mockReturnValue({
  errorMessage: '',
  successMessage: '',
  showErrorAlert: false,
  setShowSuccessAlert,
  setSuccessMessage,
  setErrorMessage,
  setShowErrorAlert,
});

jest.mock('../../../components/common/ProcessSpinner.jsx', () => () => <div data-testid='spinner' />);

test('submits email and shows success', async () => {
  const { asFragment } = render(<PasswordResetRequest />);
  fireEvent.change(screen.getByPlaceholderText(/enter email/i), { target: { value: 'me@example.com' } });
  fireEvent.click(screen.getByRole('button', { name: /send link/i }));
  const { requestPasswordReset } = require('../../../components/auth/AuthService.js');
  await waitFor(() => expect(requestPasswordReset).toHaveBeenCalledWith('me@example.com'));
  expect(setSuccessMessage).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
