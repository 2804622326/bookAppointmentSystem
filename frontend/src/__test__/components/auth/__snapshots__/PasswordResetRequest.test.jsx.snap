// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submits email and shows success 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-center container"
    style="margin-top: 100px;"
  >
    <div
      class="w-100 card"
      style="max-width: 600px;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          Password Reset Request
        </div>
        <form
          class=""
        >
          <div
            classnane="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Enter your email address
            </label>
            <input
              class="form-control"
              id="email"
              placeholder="Enter email"
              type="email"
              value=""
            />
            <small
              class="text-muted form-text"
            >
              We'll send a password reset link to your email.
            </small>
          </div>
          <button
            class="mt-2 btn btn-outline-info"
            type="submit"
          >
            Send Link
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
